src/dapp/extensionProvider.ts

```diff
@@ -0,0 +1,212 @@
+      this.popupName,
+      this.popupOptions
+    );
+  }
+
+  private startBgrMsgChannel(
+    operation: string,
+    connectData: any
+  ): Promise<any> {
+    return new Promise((resolve, reject) => {
+      window.postMessage(
+        {
+          target: "erdw-inpage",
+          type: operation,
+          data: connectData,
+        },
+        window.origin
+      );
+
+      const eventHandler = (event: any) => {
+        if (
+          event.isTrusted &&
+          event.data.type &&
+          event.data.target === "erdw-contentScript"
+        ) {
+          switch (event.data.type) {
+            case "logoutResponse":
+              window.removeEventListener("message", eventHandler);
+              resolve(true);
+              break;
+          }
+        }
+      };
+      setTimeout(() => {
+        reject(
+          "Extension logout response timeout. No response from extension."
+        );
+      }, 3000);
+      window.addEventListener("message", eventHandler, false);
+    });
+  }
+
+  private startExtMsgChannel(
+    operation: string,
+    connectData: any
+  ): Promise<any> {
+    return new Promise((resolve, reject) => {
+      var isResolved = false;
+      const eventHandler = (event: any) => {
+        if (
+          event.isTrusted &&
+          event.data.type &&
+          event.data.target === "erdw-extension"
+        ) {
+          switch (event.data.type) {
+            case "popupReady":
+              event.ports[0].postMessage({
+                target: "erdw-inpage",
+                type: operation,
+                data: connectData,
+              });
+              break;
+            case "connectResult":
+              this.extensionPopupWindow?.close();
+              this.account = event.data.data;
+              window.removeEventListener("message", eventHandler);
+              isResolved = true;
+              resolve(event.data.data);
+              break;
+            default:
+              this.extensionPopupWindow?.close();
+              window.removeEventListener("message", eventHandler);
+              isResolved = true;
+              resolve(event.data.data);
+              break;
+          }
+        }
+      };
+      var windowCloseInterval = setInterval(() => {
+        if (this.extensionPopupWindow?.closed) {
+          window.removeEventListener("message", eventHandler);
+          clearInterval(windowCloseInterval);
+          if (!isResolved)
+            reject("Extension window was closed without response.");
+        }
+      }, 500);
+
+      window.addEventListener("message", eventHandler, false);
+    });
+  }
+
+  async processTransactions(
+    transactions: Array<Transaction>,
+    signOnly: boolean
+  ): Promise<Array<Transaction>> {
+    this.openExtensionPopup();
+    const data = {
+      from: this.account.index,
+      transactions: transactions,
+      signOnly,
+    };
+
+    return await this.startExtMsgChannel("transaction", data);
```

Comment:
Actually, the same value is used for sign and send transactions, the only difference is the signOnly  flag. So changing it to signTransaction is not descriptive. I am open to suggestions.

Reply:
I would have two endpoints, I am not a big fan of having options of the type (`if flagIsSet-> doSomeOtherStuff`). This opens the door for other flags, and it can get messy. Then we'd need a refactor, which will be a breaking change and so on :)